js/WSConnection.js

```diff
@@ -4,23 +4,38 @@
  * Custom Element describing connections between worksheets
  */
 import LeaderLine from "leader-line";
+import BasicInterpreter from "./interpreters.js";
+import CommandInterface from './CommandInterface.js';
+import { EndOfStackError, CallStack } from "./callStack.js";
 
 class WSConnection extends HTMLElement {
     constructor() {
         super();
 
+        this.id;
+
+        this.interpreter = null;
+        this.callStack = null;
         this.leaderLines = [];
 
         // Bound component methods
         this.updateLeaderLine = this.updateLeaderLine.bind(this);
         this.updateLinkedSheet = this.updateLinkedSheet.bind(this);
         this.onWorksheetMoved = this.onWorksheetMoved.bind(this);
+        this.openCommandInterface = this.openCommandInterface.bind(this);
+        this.step = this.step.bind(this);
+        this.run = this.run.bind(this);
+        this.inspectCallstack = this.inspectCallstack.bind(this);
     }
 
     connectedCallback() {
         if (this.isConnected) {
+            this.setAttribute("id", window.crypto.randomUUID());
```

Comment:
This should be fine in our use case, but a new uuid will be generated any time this element gets moved around (I know that's unlikely to happen at this point). The alternative would be to set an instance property directly in the constructor and then set the attribute to that value on any connectedCallback

Reply:
this just temp to deal with callstack viewing, i am not sure how we should really identify a connection in the future whether really the callstack should carry the id etc